app/src/main/java/com/mybus/dao/RealmDao.java

```diff
@@ -0,0 +1,176 @@
+        }
+    }
+
+    /**
+     * @param id
+     * @return a copy of a realObject item finding by id
+     */
+    public T getById(Long id) {
+        return getByField("id", id);
+    }
+
+    /**
+     * @param field name of the field
+     * @param value of the field
+     * @return a copy of a realObject item finding by a field
+     */
+    public T getByField(String field, Boolean value) {
+        return mRealm.copyFromRealm(mRealm.where(mType).equalTo(field, value).findFirst());
+    }
+
+    /**
+     * @param field name of the field
+     * @param value of the field
+     * @return a copy of a realObject item finding by a field
+     */
+    public T getByField(String field, Byte value) {
+        return mRealm.copyFromRealm(mRealm.where(mType).equalTo(field, value).findFirst());
+    }
+
+    /**
+     * @param field name of the field
+     * @param value of the field
+     * @return a copy of a realObject item finding by a field
+     */
+    public T getByField(String field, Double value) {
+        return mRealm.copyFromRealm(mRealm.where(mType).equalTo(field, value).findFirst());
+    }
+
+    /**
+     * @param field name of the field
+     * @param value of the field
+     * @return a copy of a realObject item finding by a field
+     */
+    public T getByField(String field, Float value) {
+        return mRealm.copyFromRealm(mRealm.where(mType).equalTo(field, value).findFirst());
+    }
+
+    /**
+     * @param field name of the field
+     * @param value of the field
+     * @return a copy of a realObject item finding by a field
+     */
+    public T getByField(String field, Integer value) {
+        return mRealm.copyFromRealm(mRealm.where(mType).equalTo(field, value).findFirst());
+    }
+
+    /**
+     * @param field name of the field
+     * @param value of the field
+     * @return a copy of a realObject item finding by a field
+     */
+    public T getByField(String field, Long value) {
+        return mRealm.copyFromRealm(mRealm.where(mType).equalTo(field, value).findFirst());
+    }
+
+    /**
+     * @param field name of the field
+     * @param value of the field
+     * @return a copy of a realObject item finding by a field
+     */
+    public T getByField(String field, Short value) {
+        return mRealm.copyFromRealm(mRealm.where(mType).equalTo(field, value).findFirst());
+    }
+
+    /**
+     * @param field name of the field
+     * @param value of the field
+     * @return a copy of a realObject item finding by a field
+     */
+    public T getByField(String field, String value) {
+        return mRealm.copyFromRealm(mRealm.where(mType).equalTo(field, value).findFirst());
+    }
+
+    /**
+     * @param field name of the field
+     * @param value of the field
+     * @return a copy of a realObject item finding by a field
+     */
+    public T getByField(String field, Date value) {
+        return mRealm.copyFromRealm(mRealm.where(mType).equalTo(field, value).findFirst());
```

Comment:
You should create a method as copyFromRealm checking if findFirst returned null in order not to get the IllegalArgumentException, returning null as the getByField didn't match any value.
Also we can make an instrumentation test on this behavior.